### PR TITLE
Light/Dark Theme

### LINKED ISSUE #8: Light/Dark Theme Transition
We want to let the user switch between light or dark theme in the website.

Ideally this could be done without refreshing the site.

### PR DESCRIPTION
**Pull request by ErtyumPX**
Closes #8 

## What are we doing in this PR?

We are adding a new service, a kinda-wrapper that functions in every page and a component that we can put in any page that we want to let user switch the theme easily.

**Theme Service:** A general service to change the current theme, interact with `LocalService` and set the theme in local storage. It was created to easily generalise theme usage across different services and components.

**Kinda-Wrapper:** It is embedded in the `Layout.tsx` which is applied in every page if there is not a specific `Layout.tsx` written for that page. It chekcs the local storage or if the system default for the theme and applies it as the page loads.

**ToggleTheme Component:** It uses Switch component by `Tremor`, it basically switches the theme between light and dark. Uses `ThemeService` to do so. It's currently located in the top-right corner. It needs to be manually added for pages right now since we may not want it everywhere.

## How does the UI look like?

Here are some picture of the switch:

screenshot-2024-03-27_11:55:32

screenshot-2024-03-27_11:55:46

screenshot-2024-03-27_11:55:54

screenshot-2024-03-27_11:56:04

## What needs to be done later?

Setting the theme while the page loads is not very smooth, we can check it when design starts mattering a lot.

### CODE REVIEW

File: src/utils/LocalService.ts
Comment (HuseyinSimsek7904) [inline src/utils/LocalService.ts:1]: Just a simple note: please do not commit chore commits inside feature PRs. If the chore commits are not related to the feature or fix commits they just make reviewing a hell even if you have very few files and lines.

**Review by HuseyinSimsek7904 — APPROVED** [review]
Looks good!

I am sorry for blocking your progress. If you want, you can just allow yourself to push directly to the main branch since you are working in this repo by yourself. There is no need for you to wait for us to approve you.

File: src/utils/LocalService.ts
Comment (Utkub24) [reply src/utils/LocalService.ts:1]: Yep, you could push these little chore tasks directly to main. Won't be much of a problem

Comment (ErtyumPX) [reply src/utils/LocalService.ts:1]: How?

Comment (ErtyumPX) [reply src/utils/LocalService.ts:1]: :)

Comment (Utkub24) [reply src/utils/LocalService.ts:1]: :open_mouth: I added us to the ruleset bypass :)

**Review by Utkub24 — APPROVED** [review]
I believe the `/panel/auth` page is missing the toggle button, could be easily added though!
